### PR TITLE
Editor store: remove createUndoLevel and refreshPost actions

### PR DESCRIPTION
**Pull request by jsnajdr**
This PR removes two `editor` store actions that are unused. Even if someone tried to use them, they would discover they don't do anything useful at all.

`createUndoLevel` was gradually moved to the `core` store (especially in #13088), and today the `editor` reducer doesn't react to the `CREATE_UNDO_LEVEL` action at all.

`refreshPost` used to be used in the `PostPermalink` component, but that one was removed in #21099. The action used to fetch the latest version of the post from REST API, and store it in state with `resetPost`. But today the `resetPost` action resets just the `id` and `type` fields, which are kind of guaranteed to never change.

In short, gradual and mechanical refactorings removed all meaning from these two actions. Regarding backward compatibility, can we safely remove them? Given that the Gutenberg codebase doesn't use them, and anyone else doesn't have any reason to call them?

Discovered this when refreshing #35929, my PR to migrate the `editor` store to thunks.

### CODE REVIEW

**Review by adamziel — APPROVED** [review]
Good find @jsnajdr!

**Review by youknowriad — COMMENTED** [review]
Should we keep the functions with a `deprecated` call (and remove their content) instead for some versions to avoid creating JS errors for potential plugins using them even if they don't do anything?

**Comment by jsnajdr** [comment]
> Should we keep the functions with a `deprecated` call (and remove their content) instead for some versions

Yes I changed the patch to make the action creators empty (returning a `DO_NOTHING` action) and deprecated. The exact same pattern is already used for several other actions.

One thing I'm not sure about is how to express the "deprecated since" versions. Since WordPress 5.9? Since Gutenberg 13.0.0? Both versioning lines seem to be used.

**Comment by youknowriad** [comment]
I think it should be `since: "6.0"` and you can also indicate which version the function will be removed entirely like `version: "6.3"` or something.

**Comment by youknowriad** [comment]
We use Gutenberg version only for things that didn't make it yet to WordPress.

**Comment by jsnajdr** [comment]
Pushed a commit that clarifies the versions: deprecated in 6.0, removed in 6.3.

File: packages/editor/src/store/actions.js [hunk 1]
```diff
@@ -404,13 +393,16 @@ export function* undo() {
...
+ * Action that creates an undo history record.
  *
- * @return {Object} Action object.
+ * @deprecated Since Gutenberg 13.0.0
  */
 export function createUndoLevel() {
-	return { type: 'CREATE_UNDO_LEVEL' };
+	deprecated( "wp.data.dispatch( 'core/editor' ).createUndoLevel", {
+		since: '5.9',
+		alternative: 'Use the core entities store instead',
```
Comment (youknowriad) [inline packages/editor/src/store/actions.js:405]: Is GitHub missing a commit or something, I don't see the "version": "6.0"

Comment (jsnajdr) [reply packages/editor/src/store/actions.js:405]: Sorry, turns out I didn't really push the commit to the server 🙂 Please look again now.

**Review by youknowriad — APPROVED** [review]
Thanks
